fizzbuzz/src/main/java/fizz.buzz/number/factory/AbstractNumberFactory.java

```diff
@@ -0,0 +1,37 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
+
+import java.lang.reflect.InvocationTargetException;
+
+public abstract class AbstractNumberFactory implements NumberProcessor, NumberFactory {
+
+    protected NumberFactory successor;
+
+    @Override
+    public void setSuccessor(NumberFactory nextFactory) {
+        successor = nextFactory;
+    }
+
+    @Override
+    public Number build(Integer input) {
+        if (input % getDivisor() == 0) {
+            return instantiateNumber(input);
+        }
+
+        return successor.build(input);
```

Comment:
The successor is mandatory for any AbstractNumberFactory instance.
I could be enforced by setting it in the constructor

        numberFactory = new FizzBuzzNumberFactory(
                new FizzNumberFactory(
                        new BuzzNumberFactory(
                                new RegularNumberFactory()
                        )
                )
        );